hooks/terraform-fmt.sh

```diff
@@ -7,8 +7,14 @@ set -e
 # workaround to allow GitHub Desktop to work, add this (hopefully harmless) setting here.
 export PATH=$PATH:/usr/local/bin
 
-for file in "$@"; do
-  pushd "$(dirname "$file")" >/dev/null
-  terraform fmt -write=true "$(basename "$file")"
+# Store and return last failure from fmt so this can validate every directory passed before exiting
+FMT_ERROR=0
+
+for dir in $(echo "$@" | xargs -n1 dirname | sort -u | uniq); do
+  echo "--> Running 'terraform fmt' in directory '$dir'"
+  pushd "$dir" >/dev/null
+  terraform fmt -write=true || FMT_ERROR=$?
   popd >/dev/null
 done
```

Comment:
```suggestion
for file in "$@"; do
  terraform fmt -diff -check "$file" || FMT_ERROR=$?
done
```

Would you be open to simplifying this loop a bit while we're focused on making this better?

Changing this to run fmt on the parent directory means the hook will then run fmt on files not actually changed in the commit being made, which could be unexpected behavior (wouldn't matter in CI, only when a user does `git commit ...`).

The suggested loop removes the pushd/popd entirely, allowing Terraform to print the full path to the file identifying the mis-formatted file without printing a list of all files checked, accomplishing the goal of being more informative while maintaining cleaner and shorter output.

The other change I'm suggesting is using `-diff -check` so that the differences are printed (more important in CI) and the check flag so Terraform won't actually mutate the files, instead returning a proper exit code, which `-write=true` does not (instead with write=true, the pre-commit fails because of the modified files).

With the suggested change, the output would look like this:

```
Terraform fmt............................................................Failed
- hook id: terraform-fmt
- exit code: 3

main.tf
--- old/main.tf
+++ new/main.tf
@@ -34,10 +34,10 @@
     ssh = {
       ports = ["22"]
       source = {
-        field      = "ranges"
+        field = "ranges"
         value = var.firewall_ssh_source_ranges
       }
-      target_tags    = ["concourse-web", "concourse-worker"]
+      target_tags = ["concourse-web", "concourse-worker"]
     },
     web = {
       ports = ["80", "433"]
modules/servicenetworking/main.tf
--- old/modules/servicenetworking/main.tf
+++ new/modules/servicenetworking/main.tf
@@ -6,7 +6,7 @@
 resource "google_compute_global_address" "peering_address_range" {
   name          = random_id.peering_address_range.hex
   purpose       = "VPC_PEERING"
-  address_type      = "INTERNAL"
+  address_type  = "INTERNAL"
   prefix_length = var.prefix_length
   network       = var.network
 }
@@ -14,5 +14,5 @@
 resource "google_service_networking_connection" "service_networking_connection" {
   network                 = var.network
   service                 = "servicenetworking.googleapis.com"
-  reserved_peering_ranges     = [google_compute_global_address.peering_address_range.name]
+  reserved_peering_ranges = [google_compute_global_address.peering_address_range.name]
 }

```

Reply:
> Changing this to run fmt on the parent directory means the hook will then run fmt on files not actually changed in the commit being made, which could be unexpected behavior

Ah, that's a good point!

Love your suggested option. Let me try it out, but that does seem like a big improvement 👍